_posts/2023-11-17-android-platforms.md

```diff
@@ -0,0 +1,115 @@
+  multiple `android_sdk_repository` repository rules defined, you can use
+  `register_toolchains` and `--extra_toolchains` to change the order they are
+  used in.
+- `--android_crosstool_top`: Similarly to the Android SDK, the
+  `android_ndk_repository` rule exposes the NDK as a C++ toolchain, using the
+  standard toolchain resolution. You can use `register_toolchains` and
+  `--extra_toolchains` to change the selection order for these.
+
+### Can I test this out?
+
+Yes, download the latest [Bazel 7.0 release
+candidate](https://releases.bazel.build/7.0.0/rc3/index.html) (or use
+[Bazelisk](https://github.com/bazelbuild/bazelisk) with
+`USE_BAZEL_VERSION=last_rc`) and update your build flags. Please [file an
+issue](https://github.com/bazelbuild/bazel/issues/new/choose) if you encounter
+any problems.
+
+### Do I have to migrate?
+
+The legacy functionality is still present in the Bazel 7.0 LTS branch. You can
+disable the changes by passing the flag `--noincompatible_enable_android_toolchain_resolution`.
+
+If you find any issues with the new functionality, please [file an
+issue](https://github.com/bazelbuild/bazel/issues/new/choose) so that we can
+address it.
+
+However, note that the legacy functionality will be deleted in short order and
+will not be present in Bazel 8.0 when that is released.
+
+## Why are things changing?
+
+As part of the overall [platforms and
+toolchains](https://bazel.build/extending/platforms) project, all of Bazel is
+moving away from using `--cpu` flags to express target platforms, and towards
+the richer variety allowed by the `platform` rule. The current usage of `--cpu`
+is not actually as a single CPU architecture: we currently capture different
+things like "Linux on 64-bit intel" (`--cpu=k8`), "macOS on 64-bit-intel"
+(`--cpu=darwin_x86_64`), "Windows on ARM" (`--cpu=arm64_windows`): and that's
+just a few common OSes.
+
+Once we also want to describe Android and iOS devices, or custom development
+boards and their firmware, the single dimension of `--cpu` is insufficient,
+which is where the platforms concepts shine.
+
+At a more basic level, because other rules (like C++, Java, Go, and Rust) are
+moving to support platforms, Android needs to support it to, so that
+mixed-language builds work properly.
+
+## What's next for Android Rules?
+
+The biggest change coming to the Android rules is to move them out of the Bazel
+binary, and into [the independent `rules_android`
+repository](https://github.com/bazelbuild/rules_android). [Starlarkification of
+the Android Rules](https://github.com/orgs/bazelbuild/projects/17) is expected
+to be complete in Q2 2024, and users will be encouraged to migrate to the
+Starlark Android Rules between then and Bazel 8.0
+
+The current version of the Starlark Android rules also supports the same
+`--android_platforms` flag, although since these rules aren't currently fully
+ready for release, there will be breaking issues.
```

Comment:
"there will be breaking issues" raises the question, what issues?

Reply:
How about this?